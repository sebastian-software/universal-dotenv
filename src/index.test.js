import { getEnvironment, init } from "."

init()

const snapshotOpts = {
  APP_ROOT: expect.any(String),
  APP_SOURCE: expect.any(String)
}

test("Sets up process.env", () => {
  expect(process.env.APP_TEST_PATH).toBe("one/two/three.js")
  expect(process.env.TEST_IGNORE).toBe("123")
})

test("Ignores non app-specific settings", () => {
  const { raw, stringified, webpack } = getEnvironment()
  expect(raw).toMatchSnapshot(snapshotOpts)
  expect(stringified).toMatchSnapshot(snapshotOpts)
  expect(webpack).toBeDefined()
})

test("Supports manually defined envs", () => {
  process.env.APP_DATA = "yellow"
  const { raw, stringified, webpack } = getEnvironment()
  expect(raw).toMatchSnapshot(snapshotOpts)
  expect(stringified).toMatchSnapshot(snapshotOpts)
  expect(webpack).toBeDefined()
  delete process.env.APP_DATA
})

test("Supports filtering envs to export", () => {
  process.env.APP_DATA = "yellow"
  process.env.APP_DATA_JUSTME = "blue"
  const { raw, stringified, webpack } = getEnvironment({
    filter: (key) => key === "APP_DATA_JUSTME"
  })
  expect(raw).toMatchSnapshot(snapshotOpts)
  expect(stringified).toMatchSnapshot(snapshotOpts)
  expect(webpack).toBeDefined()
  delete process.env.APP_DATA
  delete process.env.APP_DATA_JUSTME
})

test("Supports translation of numbers", () => {
  process.env.APP_VERSION = 1.6
  const { raw, stringified, webpack } = getEnvironment()
  expect(raw).toMatchSnapshot(snapshotOpts)
  expect(stringified).toMatchSnapshot(snapshotOpts)
  expect(webpack).toBeDefined()
  delete process.env.APP_VERSION
})

test("Supports translation of booleans", () => {
  process.env.APP_DEBUG = true
  const { raw, stringified, webpack } = getEnvironment()
  expect(raw).toMatchSnapshot(snapshotOpts)
  expect(stringified).toMatchSnapshot(snapshotOpts)
  expect(webpack).toBeDefined()
  delete process.env.APP_DEBUG
})

test("Prevents translation of booleans when disabled", () => {
  process.env.APP_DEBUG = true
  const { raw, stringified, webpack } = getEnvironment({ translate: false })
  expect(raw).toMatchSnapshot(snapshotOpts)
  expect(stringified).toMatchSnapshot(snapshotOpts)
  expect(webpack).toBeDefined()
  delete process.env.APP_DEBUG
})

test("Exports NODE_ENV", () => {
  const { raw } = getEnvironment()
  expect(raw.NODE_ENV).toBe("test")
})

test("Exports APP_ROOT", () => {
  const { raw } = getEnvironment()
  expect(raw.APP_ROOT).toMatch(/universal-dotenv$/)
})

test("Exports APP_SOURCE", () => {
  const { raw } = getEnvironment()
  expect(raw.APP_SOURCE).toMatch(/universal-dotenv[/\\]src$/)
})

test("Adds APP_ROOT to process.env", () => {
  expect(process.env.APP_ROOT).toBeDefined()
})

test("Adds APP_SOURCE to process.env", () => {
  expect(process.env.APP_SOURCE).toBeDefined()
})
