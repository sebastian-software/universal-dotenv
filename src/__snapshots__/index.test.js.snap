// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ignores non app-specific settings 1`] = `
Object {
  "APP_ROOT": Any<String>,
  "APP_SOURCE": Any<String>,
  "APP_TEST_OVERRIDE": "overridden",
  "APP_TEST_PATH": "one/two/three.js",
  "NODE_ENV": "test",
}
`;

exports[`Ignores non app-specific settings 2`] = `
Object {
  "APP_ROOT": Any<String>,
  "APP_SOURCE": Any<String>,
  "APP_TEST_OVERRIDE": "\\"overridden\\"",
  "APP_TEST_PATH": "\\"one/two/three.js\\"",
  "NODE_ENV": "\\"test\\"",
}
`;

exports[`Prevents translation of booleans when disabled 1`] = `
Object {
  "APP_DEBUG": "true",
  "APP_ROOT": Any<String>,
  "APP_SOURCE": Any<String>,
  "APP_TEST_OVERRIDE": "overridden",
  "APP_TEST_PATH": "one/two/three.js",
  "NODE_ENV": "test",
}
`;

exports[`Prevents translation of booleans when disabled 2`] = `
Object {
  "APP_DEBUG": "\\"true\\"",
  "APP_ROOT": Any<String>,
  "APP_SOURCE": Any<String>,
  "APP_TEST_OVERRIDE": "\\"overridden\\"",
  "APP_TEST_PATH": "\\"one/two/three.js\\"",
  "NODE_ENV": "\\"test\\"",
}
`;

exports[`Supports filtering envs to export 1`] = `
Object {
  "APP_DATA_JUSTME": "blue",
  "APP_ROOT": Any<String>,
  "APP_SOURCE": Any<String>,
  "NODE_ENV": "test",
}
`;

exports[`Supports filtering envs to export 2`] = `
Object {
  "APP_DATA_JUSTME": "\\"blue\\"",
  "APP_ROOT": Any<String>,
  "APP_SOURCE": Any<String>,
  "NODE_ENV": "\\"test\\"",
}
`;

exports[`Supports manually defined envs 1`] = `
Object {
  "APP_DATA": "yellow",
  "APP_ROOT": Any<String>,
  "APP_SOURCE": Any<String>,
  "APP_TEST_OVERRIDE": "overridden",
  "APP_TEST_PATH": "one/two/three.js",
  "NODE_ENV": "test",
}
`;

exports[`Supports manually defined envs 2`] = `
Object {
  "APP_DATA": "\\"yellow\\"",
  "APP_ROOT": Any<String>,
  "APP_SOURCE": Any<String>,
  "APP_TEST_OVERRIDE": "\\"overridden\\"",
  "APP_TEST_PATH": "\\"one/two/three.js\\"",
  "NODE_ENV": "\\"test\\"",
}
`;

exports[`Supports translation of booleans 1`] = `
Object {
  "APP_DEBUG": true,
  "APP_ROOT": Any<String>,
  "APP_SOURCE": Any<String>,
  "APP_TEST_OVERRIDE": "overridden",
  "APP_TEST_PATH": "one/two/three.js",
  "NODE_ENV": "test",
}
`;

exports[`Supports translation of booleans 2`] = `
Object {
  "APP_DEBUG": "true",
  "APP_ROOT": Any<String>,
  "APP_SOURCE": Any<String>,
  "APP_TEST_OVERRIDE": "\\"overridden\\"",
  "APP_TEST_PATH": "\\"one/two/three.js\\"",
  "NODE_ENV": "\\"test\\"",
}
`;

exports[`Supports translation of numbers 1`] = `
Object {
  "APP_ROOT": Any<String>,
  "APP_SOURCE": Any<String>,
  "APP_TEST_OVERRIDE": "overridden",
  "APP_TEST_PATH": "one/two/three.js",
  "APP_VERSION": 1.6,
  "NODE_ENV": "test",
}
`;

exports[`Supports translation of numbers 2`] = `
Object {
  "APP_ROOT": Any<String>,
  "APP_SOURCE": Any<String>,
  "APP_TEST_OVERRIDE": "\\"overridden\\"",
  "APP_TEST_PATH": "\\"one/two/three.js\\"",
  "APP_VERSION": "1.6",
  "NODE_ENV": "\\"test\\"",
}
`;
